precompile_process_test.go

```diff
@@ -59,6 +75,28 @@ func testPrecompileProcess(t *testing.T, context spec.G, it spec.S) {
 			Expect(executions[0].Env).To(ContainElement("SECRET_KEY_BASE=dummy"))
 		})
 
+		it("runs the bundle exec assets:precompile process while respecting RAILS_ENV", func() {
+			os.Setenv("RAILS_ENV", "staging")
+			err := precompileProcess.Execute(workingDir)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(executions).To(HaveLen(1))
+			Expect(executions[0].Args).To(Equal([]string{"exec", "rails", "assets:precompile", "assets:clean"}))
+			Expect(executions[0].Env).To(ContainElement("RAILS_ENV=staging"))
+			Expect(executions[0].Env).To(ContainElement("SECRET_KEY_BASE=dummy"))
+		})
```

Comment:
The unit tests are currently failing, because in the test case on line 89, the `RAILS_ENV` env var is still set to `staging` from this test case. In order to separate the test cases (and to conform with how we usually set up our test cases), we should  we will do in an `it.Before(func() {...})`, and any clean up in an `it.After(func() {...})`. In this case, I think we'd want to do that with the test case below as well. We'd also want to encase the whole block where we're setting the environment variable inside of a `context` to group the before/after groups with the test case. It would look like:


```suggestion
           context("when a user sets their own RAILS_ENV", func() {
                it.Before(func() {
                  Expect(os.Setenv("RAILS_ENV", "staging")).To(Succeed())
                })
                it.After(func() {
                    Expect(os.Unsetenv("RAILS_ENV")).To(Succeed())
                })
		it("runs the bundle exec assets:precompile process while respecting RAILS_ENV", func() {
			err := precompileProcess.Execute(workingDir)
			Expect(err).NotTo(HaveOccurred())

			Expect(executions).To(HaveLen(1))
			Expect(executions[0].Args).To(Equal([]string{"exec", "rails", "assets:precompile", "assets:clean"}))
			Expect(executions[0].Env).To(ContainElement("RAILS_ENV=staging"))
			Expect(executions[0].Env).To(ContainElement("SECRET_KEY_BASE=dummy"))
		})
	})
```